physics/module_SGSCloud_RadPre.F90

```diff
@@ -131,12 +132,13 @@ SUBROUTINE sgscloud_radpre_run(            &
       ENDIF
  
      ! add boundary layer clouds
+      IF (do_mynnedmf == .true.) THEN
       do k = 1, levs
         do i = 1, im
 
-         IF( qr(i,k) > 1.0e-7 .OR. qs(i,k) > 1.0e-7)THEN
+         IF (qr(i,k) > 1.0e-7 .OR. qs(i,k) > 1.0e-7 .OR. (imfdeepcnv == imfdeepcnv_gf .AND. qci_conv(i,k)>1.0e-7) THEN
```

Comment:
@haiqinli please check that the logic I added here makes sense. Without testing for imfdeepcnv==imfdeepcnv_gf first, the code crashes when GF is not used (because qci_conv is not allocated).